From 78ee4c0bc8bc2cbcaa13983702c06aed12f1298b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 2 Jul 2018 17:37:37 +0800
Subject: [PATCH 4271/5242] MLK-19017-3 gpu: imx: add system pm support for
 LCDIF

commit  77d7047b9f234f45e4707adeebde5b3f6275b1eb from
https://source.codeaurora.org/external/imx/linux-imx.git

Implement the suspend()/resume() callbacks to support system
power management functions for LCDIF.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 7e00487012753cb370eab4ff5c05f76f7361297f)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/lcdif/lcdif-common.c |   25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/lcdif/lcdif-common.c b/drivers/gpu/imx/lcdif/lcdif-common.c
index 3ce84b5..863cb62 100644
--- a/drivers/gpu/imx/lcdif/lcdif-common.c
+++ b/drivers/gpu/imx/lcdif/lcdif-common.c
@@ -85,6 +85,20 @@ struct lcdif_soc_pdata imx8mm_pdata = {
 };
 MODULE_DEVICE_TABLE(of, lcdif_dt_ids);
 
+#ifdef CONFIG_PM
+static int imx_lcdif_runtime_suspend(struct device *dev);
+static int imx_lcdif_runtime_resume(struct device *dev);
+#else
+static int imx_lcdif_runtime_suspend(struct device *dev)
+{
+	return 0;
+}
+static int imx_lcdif_runtime_resume(struct device *dev)
+{
+	return 0;
+}
+#endif
+
 void disp_mix_bus_rstn_reset(struct regmap *gpr, bool reset)
 {
 	if (!reset)
@@ -614,11 +628,20 @@ static int imx_lcdif_remove(struct platform_device *pdev)
 #ifdef CONFIG_PM_SLEEP
 static int imx_lcdif_suspend(struct device *dev)
 {
-	return 0;
+	return imx_lcdif_runtime_suspend(dev);
 }
 
 static int imx_lcdif_resume(struct device *dev)
 {
+	return imx_lcdif_runtime_resume(dev);
+}
+#else
+static int imx_lcdif_suspend(struct device *dev)
+{
+	return 0;
+}
+static int imx_lcdif_resume(struct device *dev)
+{
 	return 0;
 }
 #endif
-- 
1.7.9.5

